trio/_core/_run.py

```diff
@@ -2373,8 +2526,7 @@ def unrolled_run(
                     # more Context.run adds.
                     tb = task_exc.__traceback__
                     for _ in range(1 + CONTEXT_RUN_TB_FRAMES):
-                        if tb is None:
-                            break
+                        assert tb is not None
```

Comment:
Next traceback can be None in some cases, I think this check is important

Reply:
I thought I added this check, and it was tripping coverage. So I should revert.